### PR TITLE
add feature/fbfw-79-axios

### PR DESCRIPTION
**Pull request by nebevets**
this pr adds `axios` and `@types/axios` (for typescript) to the project. existing api routes are defined as a enum called `ApiRoute`. in order to use this branch, you must run `npm i`

two helper functions are added to help make `GET` requests:

`buildQueryString`
---
takes an object of key value pairs and reduces them to a query string.

`getGameApiRequest`
---
takes a route (one of `ApiRoute`) and an object of key values pairs for the query string (built by `buildQueryString`) and makes a get request.

in order to properly use this, you can set up three files in your `beta-src` folder:

1. `.env.local`
2. `.env.production`
3. `.env.test`

---
for local testing,`.env.local` is used and it contains the following information: 
```
REACT_APP_WD_API_BASE_URL=http://localhost/api.php
```

you must run `npm run build-local` and test the api from `localhost` not `localhost:3000`. if you test from `localhost:3000` your request is going to be denied by CORS policy.

---
for the test environment,`.env.test` is used and it contains the following information: 
```
REACT_APP_WD_API_BASE_URL=https://test.webdiplomacy.net/api.php
```

then, `npm run build-test` and test the api from `test.webdiplomacy` not `localhost:3000`.

---
for the production environment,`.env.production` is used and it contains the following information: 
```
REACT_APP_WD_API_BASE_URL=https://webdiplomacy.net/api.php
```

then, `npm run build` and test the api from `test.webdiplomacy` not `localhost:3000`.

---

NOTE: testing the api is going to be done in a following PR, TBD.

### CODE REVIEW

**Review by bryancodazen — CHANGES_REQUESTED** [review]
Left a small change request and question.

File: beta-src/src/utils/api/index.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      return keyValuePairs;
+    }, [])
+    .join("&");
+
+const api = axios.create({
+  baseURL: process.env.REACT_APP_WD_API_BASE_URL,
+  headers: {
+    "Content-Type": "multipart/form-data",
+  },
+});
```
[the remark this answers is not in the document]
Comment (nebevets) [reply beta-src/src/utils/api/index.ts:23]: @bryancodazen i wasn't sure what headers we'll need, so i just added this boilerplate to show how it's done for all headers that are to be common. if we want different headers for a different api request, we can add them to a new function made specifically for that api request and they will be in addition to whatever is common.

**Review by sariacodazen — APPROVED** [review]
LGTM

**Review by bryancodazen — APPROVED** [review]
LGTM

**Comment by shelbycodazen** [comment]
LGTM :shipit:

**Pull request merged**